### DISCUSSION TITLE
Get client language in getInitialProps method

### DISCUSSION THREAD

**Discussion by nueq22** [vinissimus/next-translate#571] [Q&A] [answered]
I need to set HTTP header with client locale in ```_app.js => getInitialProps```. 

How I can get the current i18n locale in this method? 

Previously I used [this lib](https://github.com/isaachinman/next-i18next) that provides me something like this
 ```  
 App.getInitialProps = async ({ Component, ctx, ...rest }) => {
   const currentLanguage = isClient ? i18n.language : ctx.req.i18n.language;
...
}
```

How I can do something similar with the current package?

**Comment by aralroca** [comment, marked as answer]
I think you will have to wait until this PR is merged on Next.js core: https://github.com/vercel/next.js/pull/21930

**Reply by nueq22** [reply]
Oh, exactly what I was looking for. Thank you for the answer. For now, I use ```router.locale```, seems that result is pretty similar to what I want
